app/main/views/questionnaire.py

```diff
@@ -21,46 +14,11 @@ def questionnaire():
 
     logger.debug("Current user %s", current_user)
 
-    eq_id = current_user.get_eq_id()
-    form_type = current_user.get_form_type()
-    logger.debug("Requested questionnaire %s for form type %s", eq_id, form_type)
-
-    schema = _load_and_parse_schema(eq_id, form_type)
-    if not schema:
-        return errors.page_not_found()
-
-    # load the response store
-    response_store = factory.create("response-store")
-
-    # load the validation store
-    validation_store = factory.create("validation-store")
-
-    # Create the validator
-    validator = Validator(schema, validation_store, response_store)
-
-    # Create the routing engine
-    routing_engine = RoutingEngine(schema, response_store)
-
-    # load the navigation history
-    navigation_history = factory.create("navigation-history")
-
-    # create the navigator
-    navigator = Navigator(schema, navigation_history)
-    if navigator.get_current_location() is None:
-        navigator.go_to('questionnaire')
-
-    # instantiate the questionnaire manager
-    questionnaire_manager = QuestionnaireManager(schema,
-                                                 response_store,
-                                                 validator,
-                                                 validation_store,
-                                                 routing_engine,
-                                                 navigator,
-                                                 navigation_history)
+    questionnaire_manager = create_questionnaire_manager()
 
     if request.method == 'POST':
         questionnaire_manager.process_incoming_responses(request.form)
-        current_location = navigator.get_current_location()
+        current_location = questionnaire_manager.navigator.get_current_location()
```

Comment:
I wonder if it would be better to hide the navigator behind the questionnaire manager. So rather than do this, create a method on questionnaire manager called get_current_location() and then delegate the call through to the navigator component. This hides its internals away then.